olp-cpp-sdk-dataservice-read/src/CatalogClientImpl.cpp

```diff
@@ -124,6 +125,39 @@ CancellableFuture<CatalogVersionResponse> CatalogClientImpl::GetLatestVersion(
   return client::CancellableFuture<CatalogVersionResponse>(
       std::move(cancel_token), std::move(promise));
 }
+
+client::CancellationToken CatalogClientImpl::ListVersions(
+    VersionsRequest request, VersionsResponseCallback callback) {
+  auto schedule_get_versions_list = [&](VersionsRequest request,
+                                        VersionsResponseCallback callback) {
+    auto catalog = catalog_;
+    auto settings = settings_;
+    auto pending_requests = pending_requests_;
+
+    auto versions_list_task =
+        [=](client::CancellationContext context) -> VersionsResponse {
+      return repository::CatalogRepository::GetVersionsList(
+          std::move(catalog), context, std::move(request), std::move(settings));
```

Comment:
I wonder why do we need then copy parameters on CatalogRepository::GetVersionsList() if it is sync? Can we move towards pass by reference to save copies?

Reply:
Could we do this in separate pull request and make this changes also in other clients?